FRW-8550 Dynamic Multistore documentation. (#2919)
spryker/spryker-docs: _includes/pbc/all/install-features/202410.0/enable-dynamic-multistore.md

  * Backoffice
  * MerchantPortal
  * Console Commands
  * Gateway
  * BackendAPI

2. Update your custom console commands  (if any) to meet the following rules:
- `Store(Facade|Client)::getCurrentStore()` should never be used in the code that console command executes.
- All store aware commands should implement `Spryker\Zed\Kernel\Communication\Console\StoreAwareConsole` and execute actions for specific store if store parameter is provided, or for all the stores in the region otherwise.
- It is recommended to use `--store` parameter instead of `APPLICATION_STORE` env variable, despite the support of env variable still exists.
3. Be aware that after enabling Dynamic Multistore mode, your basic domain structure will change from store to region for all the applications(Example https://yves.de.mysprykershop.com => https://yves.eu.mysprykershop.com), make sure that it is expected. If external systems are impacted by this - necessary redirects are set, so SEO of your site is not impacted.

> 4. The Dynamic Store feature itself does not require any database changes, in case you've already migrated to the latest demoshop version.

Reviewer: Who is this information useful for?

Author: For the developer who enabled DMS ON mode

5. Dynamic Multistore introduce some changes in RabbitMQ messages structure, so it is **important** that: 
   - During the migration we do not have unprocessed messages in the queue. Make sure that all messages are processed **before** enabling Maintenance Mode.
   - Make sure that `Maintainance Mode` is enabled during migration to make sure that no new messages are added to the queue before the migration is finished.

>    (Expected downtime is limited to the deployment time, normally it takes less than 1hr)

Reviewer: Downtime of what?

> This document describes how to enable [Dynamic Multistore](/docs/pbc/all/dynamic-multistore/{{page.version}}/base-shop/dynamic-multistore-feature-overview.html) on the latest codebase.

Reviewer: Add front matter

> This document describes how to enable [Dynamic Multistore](/docs/pbc/all/dynamic-multistore/{{page.version}}/base-shop/dynamic-multistore-feature-overview.html) on the latest codebase.

Reviewer: What do you mean by the latest codebase?

## Enable Dynamic Multistore

{% info_block warningBox "Project version" %}

> If your project version is below 202307.0, you need to update to the latest codebase first, see [Install Dynamic Multistore](/docs/pbc/all/dynamic-multistore/202410.0/base-shop/install-dynamic-multistore.html) page.

Reviewer: What do I need to do with the linked document?

Author: Rephrased, please take a look.
I would prefer not to merge those documents, this is a different things.

{% endinfo_block %}

### Here are the steps that needs to be performed to enable Dynamic Multistore.

{% info_block warningBox "Staging environment" %}

>      Make sure that **all** steps above are performed (and fully tested) on staging before applying it on production setup, to avoid unexpected downtime and data loss.

Reviewer: Perhaps you wanted to say "steps below"?

{% endinfo_block %}

> 1. Make sure that your custom Backoffice, MerchantPortal, Console Commands, Gateway, BackendAPI code do not use `StoreFacade::getCurrentStore()` as well as `StoreClient::getCurrentStore()` methods, they are no longer available in any other application except GlueStorefront and Storefront. 

Reviewer 2: It would be good to move https://docs.spryker.com/docs/pbc/all/dynamic-multistore/202410.0/base-shop/dynamic-multistore-feature-overview.html#how-dynamic-multistore-affects-a-project here and leave reference at https://docs.spryker.com/docs/pbc/all/dynamic-multistore/202410.0/base-shop/dynamic-multistore-feature-overview.html to docs/pbc/all/dynamic-multistore/202410.0/base-shop/difference-between-modes.md

> 1. Make sure that your custom Backoffice, MerchantPortal, Console Commands, Gateway, BackendAPI code do not use `StoreFacade::getCurrentStore()` as well as `StoreClient::getCurrentStore()` methods, they are no longer available in any other application except GlueStorefront and Storefront. 

Reviewer: ```suggestion
1. `StoreFacade::getCurrentStore()` and `StoreClient::getCurrentStore()` methods are available only in GlueStorefront and Storefront applications. Replace them with other methods in the following entities: 
  * Backoffice
  * MerchantPortal
  * Console Commands
  * Gateway
  * BackendAPI
```

> 2. Make sure that you custom console commands (and the places where they are executed) updated to the new format, see  [details](/docs/pbc/all/dynamic-multistore/{{page.version}}/base-shop/difference-between-modes.html#Deployment file difference).

Reviewer: Explain how to do it in this document, jumping between documents is not acceptable. Same for step 5

> 3. Be aware that after enabling Dynamic Multistore mode, your basic domain structure will change from store to region, make sure that it is expected. If external systems are impacted by this - necessary redirects are set, so SEO of your site is not impacted.

Reviewer 2: Rephrase please to clarify that subdomain part changes 

`your basic domain structure will change from store to region`

4. The Dynamic Store feature itself does not require any database changes, in case you've already migrated to the latest demoshop version.
5. Dynamic Multistore introduce some changes in RabbitMQ messages structure, so it is **important** that: 

>    - During server migration we do not have unprocessed messages in the queue. Make sure that all messages are processed **before** enabling Maintenance Mode.

Reviewer 3: I think you talk to "you" not "we"

>    - During server migration we do not have unprocessed messages in the queue. Make sure that all messages are processed **before** enabling Maintenance Mode.

Reviewer: How is server migration related to dynamic multistore?

   - Make sure that `Maintainance Mode` is enabled during migration to make sure that no new messages are added to the queue before the migration is finished.
   (Expected downtime is limited to the deployment time, normally it takes less than 1hr)
5. Update deployment your AWS deployment files to Dynamic Multistore mode, see  [details](/docs/pbc/all/dynamic-multistore/{{page.version}}/base-shop/difference-between-modes.html#Difference in console commands execution).
6. Run normal deploy for your server pipeline.

{% info_block warningBox "Verification" %}
- Make sure your store is accessible at `https://yves.eu.mysprykershop.com` or `https://backoffice.eu.mysprykershop.com`.
- Make sure the store switcher is displayed on the Storefront.

Congratulations! Now you have Dynamic Multistore feature up and running.
{% endinfo_block %}

> ## Rollout 

Reviewer 3: Rollback? I think you use the wrong word.
